rueidis.go

```diff
@@ -77,6 +77,10 @@ type ClientOption struct {
 	Password   string
 	ClientName string
 
+	// AuthCredentialsFn allows for setting the AUTH username and password dynamically on each connection attempt to
+	// support rotating credentials
+	AuthCredentialsFn func() (username string, password string, err error)
```

Comment:
Great suggestion, and just so I understand correctly, at this point would you recommend `AuthCredentialsContext` be an empty struct so that new fields can be added as needed, or are there some fields that should be incorporated into `AuthCredentialsContext` at the moment?

Reply:
An empty struct is okay for now. But I think adding a `conn.RemoteAddr()` into it may be helpful. For example, people can return different `AuthCredentials` based on the connecting address.